Trippy/Utils/MapUtils.m

```diff
@@ -35,11 +35,18 @@ + (UIImage *)getStaticMapImage:(CLLocationCoordinate2D)location width:(int)width
 }
 
 + (NSString *)generateOptimizedDirectionsApiUrl:(LocationCollection *)collection
-                                          origin:(Location *)origin
+                                         origin:(Location *)origin
+                                  omitWaypoints:(NSArray *)omitWaypoints
                                    departureTime:(NSDate *)departureTime {
     NSString *stops = @"optimize:true";
-    for(Location *loc in collection.locations) {
+    int count = 0;
+    for (Location *loc in collection.locations) {
+        if ([omitWaypoints containsObject:@(count)]) {
+            count += 1;
```

Comment:
This and line 49 can be moved to before the if check, since it happens in all cases.